packages/medusa/src/repositories/order.ts

```diff
@@ -25,7 +25,8 @@ export class OrderRepository extends Repository<Order> {
       Object.entries(groupedRelations).map(async ([_, rels]) => {
         return this.findByIds(entitiesIds, {
           select: ["id"],
-          relations: rels as string[],
+          relations: rels,
+          withDeleted: rels.includes("region"),
```

Comment:
Wouldn't it be:
```
 Object.entries(groupedRelations).map(async ([rel, rels]) => {
        return this.findByIds(entitiesIds, {
          select: ["id"],
          relations: rels,
          withDeleted: rel === "region"
```

I believe `rels` will be all the additional relations added on `region` no?

Reply:
The passed relations: `["region", "region.[something]", "region.[something_else]"]`

Reply:
I did the changes using topLevel but it is the same